### PR TITLE
Make Testing Optional

### PR DESCRIPTION
**Pull request by MixusMinimax**
Hey, I don't use Visual Studio, so I don't have the testing framework set up. (I'm using CPM to use this project in my own)

So here I make testing optional (ON by default), because otherwise ctest won't work with my other tests.

Then, you can simply do the following (for example):
```cmake
CPMAddPackage(
        NAME inipp
        VERSION 1.0.13
        GITHUB_REPOSITORY MixusMinimax/inipp
        GIT_TAG 1.0.13
        OPTIONS "INIPP_ENABLE_TESTING OFF"
)
```

### CODE REVIEW

**Comment by mcmtroffaes** [comment]
Thanks for the contribution. It could be very useful to disable building of the test suite indeed. There's already the ``BUILD_TESTING`` variable from the CTest module, which gives a more "canonical" solution if you like, e.g.

```cmake
include(CTest)
if(BUILD_TESTING)
  add_subdirectory("unittest/")
endif()
```

See https://cmake.org/cmake/help/latest/command/enable_testing.html

Would that work for you too?

**Comment by MixusMinimax** [comment]
Ah yes, even better!

File: CMakeLists.txt [hunk 1]
```diff
@@ -4,5 +4,7 @@ set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 include_directories("inipp/")
 install(FILES inipp/inipp.h DESTINATION include)
-enable_testing()
-add_subdirectory("unittest/")
+if (BUILD_TESTING)
+    enable_testing()
```
Comment (mcmtroffaes) [inline CMakeLists.txt]: BUILD_TESTING is only set when CTest is included. Also CTest will call enable_testing. Does this work for you?

```suggestion
include(CTest)
if(BUILD_TESTING)
```

**Review by mcmtroffaes — CHANGES_REQUESTED** [review]

**Comment by mcmtroffaes** [comment]
LGTM, thanks for the contribution!

File: CMakeLists.txt (hunk 1 above)
Comment (MixusMinimax) [reply CMakeLists.txt]: Oups, it seems like we both did the same thing at the same time

**Comment by MixusMinimax** [comment]
Thanks for your help and feedback!

File: CMakeLists.txt (hunk 1 above)
Comment (mcmtroffaes) [reply CMakeLists.txt]: LOL yes 😄

**Pull request merged**